### PR TITLE
[BE] Add authentication models, serializers, views

### PR DESCRIPTION
**Pull request by jaydonlau**
Groundwork for the login/registration flow:

## New packages
- installed djangorestframework and djangorestframework-jwt

## Description
- Adds a custom user model - even though we aren't adding any custom functionality, django recommends using a custom user model because it affects later migrations.
- Adds a user profile to contain user data, and serializers for both models
- Adds a login and register api

todo: add user profile api

Took most of the code from this tutorial: https://medium.com/analytics-vidhya/django-rest-api-with-json-web-token-jwt-authentication-69536c01ee18

key differences is that we aren't using a user manager, and signing in is based on username and not email (can change in the future)

## Features
- register api at `http://localhost:8000/users/register/` with the following request body:
```
{
	"username": "blah",
	"password":"123456",
	"profile": {
		"first_name": "name",
		"last_name": "name2",
		"phone_number": "1234567891",
		"age": 20,
		"gender": "M"
    }
}
```

- login api at `http://localhost:8000/users/login/` with the following request body: 
```
{
	"username": "blah",
	"password":"123456",
}
```
comes back with response:
```
{
    "success": "True",
    "status code": 200,
    "message": "User logged in  successfully",
    "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoxLCJ1c2VybmFtZSI6ImNocm9zbHkiLCJleHAiOjE1ODQ3NjYxNzgsImVtYWlsIjoiY2hyb3NseUBnbWFpbC5jb20ifQ.7f5L18Zl4e2r_iENqJR3IA3yBe7NFmPEcfPev28uRvk"
}
```
we can specify endpoints with JWT authentication that will require a valid JWT token from the login API

### CODE REVIEW

File: backend/sic_project/apps/auth/users/models.py [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    user = models.OneToOneField(CustomUser, on_delete=models.CASCADE, related_name='profile')
+    first_name = models.CharField(max_length=50, unique=False)
+    last_name = models.CharField(max_length=50, unique=False)
+    phone_number = models.CharField(max_length=10, unique=True, null=False, blank=False)
+    age = models.PositiveIntegerField(null=False, blank=False)
+    GENDER_CHOICES = (
+        ('M', 'Male'),
+        ('F', 'Female'),
+    )
```
Comment (denzelkwan) [inline backend/sic_project/apps/auth/users/models.py]: niceee. I like the only two options

File: backend/sic_project/apps/auth/users/views.py [hunk 2]
```diff
@@ -1,3 +1,42 @@
...
+from rest_framework import status
+from rest_framework.generics import RetrieveAPIView, CreateAPIView
+from rest_framework.response import Response
+from rest_framework.permissions import AllowAny
+from apps.auth.users.serializers import UserLoginSerializer, UserRegistrationSerializer
 
-# Create your views here.
+class UserLoginView(RetrieveAPIView):
+
+    permission_classes = (AllowAny,)
```
Comment (denzelkwan) [inline backend/sic_project/apps/auth/users/views.py]: what's the trailing comma doing?

Comment (jaydonlau) [reply backend/sic_project/apps/auth/users/views.py]: nothing, python lets you use trailing commas for data tuples and list so diffs are easier and minor comma bugs are avoided when the entries are separated in different lines: https://stackoverflow.com/questions/11597901/why-are-trailing-commas-allowed-in-a-list

short answer is i copy pasted the code

File: backend/sic_project/apps/auth/users/models.py (hunk 1 above)
Comment (jaydonlau) [reply backend/sic_project/apps/auth/users/models.py]: :s

**Pull request merged**